src/workerd/api/html-rewriter.c++

```diff
@@ -516,14 +519,15 @@ kj::Promise<void> Rewriter::flushWrite() {
     return kj::READY_NOW;
   };
 
-  KJ_IF_SOME(wp, writePromise) {
-    KJ_DEFER(writePromise = kj::none);
-    return wp.then([checkException]() { return checkException(); });
+  if (!outputBuffer.empty()) {
+    auto bufferCopy = kj::heapArray(outputBuffer.asPtr());
+    outputBuffer.clear();
+
+    return inner->write(bufferCopy).then([checkException]() { return checkException(); });
```

Comment:
I think this is a UaF, because we don't attach `bufferCopy` to the returned promise.

But, I think we probably don't have to make a copy anyway, since we don't expect anyone to add more data to the buffer until we finish flushing it here. I also think we could convert this function to a coroutine to simplify it a bit (it was written before we implemented KJ coroutines). Something like this:

```cpp
kj::Promise<void> Rewriter::flushWrite() {
  if (!outputBuffer.empty()) {
    KJ_DEFER(outputBuffer.clear());
    co_await inner->write(outputBuffer);
  }
  KJ_IF_SOME(exception, maybeException) {
    inner->abort(kj::cp(exception));
    kj::throwFatalException(kj::cp(exception));
  }
}
```